src/components/slide-toggle/slide-toggle.ts

```diff
@@ -84,6 +92,23 @@ export class MdSlideToggle implements AfterContentInit, ControlValueAccessor {
     this._isInitialized = true;
   }
 
+  /** TODO: internal */
+  ngAfterViewInit() {
+    let hammerManager = new Hammer(this._thumbContainerRef.nativeElement);
+
+    // The Pan / Drag recognition should immediately detect the starting drag event, without
+    // waiting for the drag to exceed the elements width.
+    hammerManager.add(new Hammer.Pan({
+      threshold: 0,
+      pointers: 0,
+      direction: Hammer.DIRECTION_HORIZONTAL
+    }));
+
+    hammerManager.on('panstart', this._onDragStart.bind(this));
+    hammerManager.on('panmove', this._onDrag.bind(this));
+    hammerManager.on('panend', this._onDragEnd.bind(this));

```

Comment:
Why are you adding the events directly instead of using handlers defined via the template, e.g.,
```html
<div class="md-slidetoggle-thumb" (dragstart)="..." ...>
```
?

Check out @iveysaur's PR #779 for the slider; she's using the baked-in gesture stuff